### ISSUE TITLE
Fusio pub/sub not working

### ISSUE THREAD

**Issue by anshulupreti** [apioo/fusio#390]
I am unable to figure out how to get the pub/sub setup. 

I have created a route for the event to be executed in real time and send out the push notification to the frontend application but this doesnt seem to be working. Following are the details of what I have done vs what I am trying to accompolish

Routes.php

/order_call: This will be the endpoint to which the frontend app will subscribe to. This route is set to resolve to a php file that pulls out the data.

subscription.php: This will be the file pulling the data. This file has the dispatcher event in the format:

$this->dispatcher->dispatch('subscribe_v1', [
    'foo' => 'bar'
]);

Now when I try to subscribe to the app with the following code, it gives me authorization denied error:

$.ajax({
    type: "POST",
  url: "https://blahblah.com/fusio/public/consumer/subscription",
  dataType: 'json',
  headers: {
    "Authorization": "Basic " + btoa(username + ":" + password)
  },
  data:  { 
    "username": "xxx",
    "password": "xxxxxxxxxx",
    "event": "subscription",
    "endpoint": "https://blahblah.com/fusio/public/index.php/order_call/"
  },
  success: function (result){
    console.log(result)
  }
});

Any help would be highly appreciated!

**Comment by chriskapp** [comment]
Hi, so you need to activate the cronjob system s. https://fusio.readthedocs.io/en/latest/concept/cronjob.html which then enables Fusio to execute all events at the background. Regarding your post call, you need to provide an `Bearer` token at the authorization header which you can obtain at the `/consumer/login` endpoint s.
https://demo.fusio-project.org/apps/internal/#!/api/consumer/login

**Comment by anshulupreti** [comment]
Thank you so much! I was successfully able to subscribe and get the response as Subscription created successfully. 

However, the subscriber dispatch event's does not send back the data to the listener like in your example with 
"foo" => "bar"

I just get the subscription successfully created response and that's all. Even though the CRON is running through the backend app and everything seems to be mapped properly. Any idea why?

**Comment by anshulupreti** [comment]
Hi Chris, 

Please let me know what to do here?

**Comment by chriskapp** [comment]
Hi, mhhh could you check whether the table `fusio_event_trigger` contains an entry containing the payload which you want to send via the event.

**Comment by anshulupreti** [comment]
Yes it does. here is the format: {"specversion":"1.0","type":"org.fusio-project.event.create","source":"\/backend\/event\/subscription","id":"86f029dc-a025-47d9-9991-12db839c57f3","datacontenttype":"application\/json","data":{"id":15,"eventId":43,"userId":4,"endpoint":"https:\/\/xxxxxxxx.in\/fusio\/public\/index.php\/order_call"}}

**Comment by anshulupreti** [comment]
Please suggest what can be done. If you have some support channel on paid basis, I can use that too.

**Comment by anshulupreti** [comment]
Hi Chris, thanks. Yes I think you got it right. 

I will help you in understanding this better. by telling you my use case.

1. I have a frontend application (client) running on angular. Let's call it A. 
2. I then have an external third party application that is gathering, saving & distributing the core data it has. Let's call it B. 
3. In the middle of the A & B is sitting my middleware built with fusio. Let's call it P. 

P is used to save the local copy of the data sent to B and to provide endpoints to receive the response or updated information. This part works absolutely fine.

B also sends real time notifictions/calls whenever an external action is generated. And since B can send multiple requests any hour/section or minute, A needs to keep subscribed to P and wait for a state change or data stack whenever a new event notification is sent from B to P. Since A is a front end app (angular) it cannot have its own endpoints and relying on Fusio to mediate the data transaction between  B to A.

This is what is happening right now or needs to happen:

1. A, (my frontend app) subscribes to an event called subscribe_v1 and it works well. 
2. The subscription event is called through an enpoint called order_call which invokes an action.
3. this action contains the standart fetch codes to receive the latest data from B. And then. the received data from B needs to dispatched to listens while storing some info in internal fusio datanase.

This is the outline of what needs to happen with the real time ordering section. I hope you haunder stpod. of amy questions, pleqe tell.accpreciate

**Comment by anshulupreti** [comment]
Hi Chris, 

Any comments on this?
